fs/fcb2/src/fcb_getprev.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include <stddef.h>
+
+#include "fcb/fcb.h"
+#include "fcb_priv.h"
+
+static int
+fcb_sector_find_last(struct fcb *fcb, struct fcb_entry *loc)
+{
+    int rc;
+    int last_valid = 0;
+
+    loc->fe_entry_num = 1;
+
+    do {
+        rc = fcb_elem_info(loc);
+        if (rc == 0) {
+            last_valid = loc->fe_entry_num;
+        }
+        if (rc == FCB_ERR_NOVAR) {
+            /*
+             * Out entries in this sector
+             */
+            if (last_valid == 0) {
+                loc->fe_entry_num = 1;
+                /*
+                 * No valid entries in this sector.
+                 */
+                return FCB_ERR_NOVAR;
+            } else {
+                loc->fe_entry_num = last_valid;
+                rc = fcb_elem_info(loc);
+                assert(rc == 0); /* must be; just succeeded a bit earlier */
+                return rc;
+            }
+        }
+        loc->fe_entry_num++;
+    } while (1);
```

Comment:
do { } while (1) is less frequent in mynewt then just while (1) { }. This PR already uses while (1) {} in other commit. Nothing wrong just a glitch in the Matrix.

Reply:
Indeed. Obviously code must've started as something else. Will change this.

Reply:
there were two places :), the other one get the fix